### PR TITLE
Support edge case where proto.Equal received a message with casttype for bytes.

### PR DESCRIPTION
**Pull request by prestonvanloon**
See the following newly failing test which is now resolved with these changes. 

```
--- FAIL: TestEqual (0.00s)
panic: interface conversion: interface {} is casttype.Bytes, not []uint8 [recovered]
	panic: interface conversion: interface {} is casttype.Bytes, not []uint8

goroutine 34 [running]:
testing.tRunner.func1(0xc0000d0800)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x6d71a0, 0xc0002962d0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/gogo/protobuf/proto.equalAny(0x6d3e40, 0x98f9c0, 0x197, 0x6d3e40, 0x98faa0, 0x197, 0xc0002a4c00, 0x6d3e40)
	/tmp/protobuf/proto/equal.go:216 +0x1a06
github.com/gogo/protobuf/proto.equalStruct(0x719b20, 0x98f980, 0x199, 0x719b20, 0x98fa60, 0x199, 0x445400)
	/tmp/protobuf/proto/equal.go:114 +0x3cd
github.com/gogo/protobuf/proto.Equal(0x778be0, 0x98f980, 0x778be0, 0x98fa60, 0x0)
	/tmp/protobuf/proto/equal.go:92 +0x347
github.com/gogo/protobuf/proto_test.TestEqual(0xc0000d0800)
	/tmp/protobuf/proto/equal_test.go:247 +0x12c
testing.tRunner(0xc0000d0800, 0x73d8c8)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x353
```

### CODE REVIEW

**Comment by prestonvanloon** [comment]
What can I do to get a PR review?

**Comment by prestonvanloon** [comment]
Any chance for a review?

**Comment by austinfromboston** [comment]
👍 would be great to get this into the next version

**Comment by jmarais** [comment]
Hi.

We can try and get it into the next patch version.



Regards,
Jacques Marais

File: proto/equal_test.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ import (
 	. "github.com/gogo/protobuf/proto"
 	proto3pb "github.com/gogo/protobuf/proto/proto3_proto"
 	pb "github.com/gogo/protobuf/proto/test_proto"
+	casttypepb "github.com/gogo/protobuf/test/casttype/combos/both"
```
Comment (jmarais) [inline proto/equal_test.go]: I would rather not import /test from the /proto package. I would rather have a stand-alone test in the /test package.

I didn't find any other casttype tests. So lets create a new package at:
./test/casttype/singles
We can then create specific casttype tests in that package which the 'combos' package doesn't cover.

So lets define a singles.proto with a message containing the ./test/castype.Bytes as a casttype.

Then we can just create our equal test; Instantiating two message with bytes casttypes and calling proto.Equal with them.

When defining a makefile for the ./test/casttype/singles/ package, make sure to use the right number of ```../``` for the protoc paths. And add the regeneration to the project root makefile.

File: proto/equal.go [hunk 2]
```diff
@@ -202,6 +202,17 @@ func equalAny(v1, v2 reflect.Value, prop *Properties) bool {
 			if v1.IsNil() != v2.IsNil() {
 				return false
 			}
+
+			// Edge case: if field is a cast type of []byte, special conversion
+			// to []byte is required for bytes.Equal.
+			if v1.Type() != reflect.TypeOf([]byte{}) {
```
Comment (jmarais) [inline proto/equal.go]: We can use the prop here to check if we are handling a casttype:
```len(prop.CastType) > 0```
And then since we know we are handling bytes. Would:
v1.Bytes()
work?
